code/modules/events/egg_hunt.dm

```diff
@@ -0,0 +1,53 @@
+/datum/event/egg_hunt
+	startWhen = 30
+	endWhen = 900
+
+/datum/event/egg_hunt/setup()
+	var/datum/announcement/centcomm/egghunt/pre/announcement_pre = new
+	announcement_pre.play()
+
+/datum/event/egg_hunt/start()
+	var/eggsspawned = 0
+	var/eggsmax = player_list.len * 8 // 8 eggs per player
+	for(var/i in 1 to 3000) // 3000 attempts
+		var/turf/candidate = locate(rand(1, world.maxx), rand(1, world.maxy), pick(SSmapping.levels_by_trait(ZTRAIT_STATION)))
+		if(isfloorturf(candidate))
+			var/hasCloset = FALSE;
+			for(var/obj/structure/closet/C in candidate)
+				new /obj/random/foods/boiledegg(C);
+				hasCloset = TRUE;
+			if(!hasCloset)
+				new /obj/random/foods/boiledegg(candidate);
+			eggsspawned += 1;
```

Comment:
```suggestion
```suggestion
			var/obj/structure/closet/C = locate() in candidate
			if(C)
				new /obj/random/foods/boiledegg(C)
			else
				new /obj/random/foods/boiledegg(candidate)
			eggsspawned += 1
```

тут откуда-то еще ; взялись